src/servers/simple.cc

```diff
@@ -111,11 +111,25 @@ ResponseAlloc(
       allocated_ptr = malloc(byte_size);
 #ifdef TRTIS_ENABLE_GPU
     } else if (use_gpu_memory) {
-      auto err = cudaMalloc(&allocated_ptr, byte_size);
+      int current_device;
+      auto err = cudaGetDevice(&current_device);
```

Comment:
We should not push this requirement into the user callback code. Instead in the provider where we invoke this callback we should remember the current device and set it back after returning from the callback. I think get and set take about the same time so we should just set the device on return and not bother to check if it has changed.